systems/citybuilding.go

```diff
@@ -46,29 +53,49 @@ func (cb *CityBuildingSystem) New(w *ecs.World) {
 // Update is ran every frame, with `dt` being the time
 // in seconds since the last frame
 func (cb *CityBuildingSystem) Update(dt float32) {
-	if engo.Keys.Get(engo.F1).JustPressed() {
-		fmt.Println("The gamer pressed F1")
+	keyQ := engo.Button{[]engo.Key{engo.Q}, "Q"}
 
-		city := City{BasicEntity: ecs.NewBasic()}
+	if keyQ.JustPressed() {
+		fmt.Println("The gamer pressed Q")
 
-		city.SpaceComponent = engo.SpaceComponent{
-			Position: engo.Point{cb.mouseTracker.MouseComponent.MouseX, cb.mouseTracker.MouseComponent.MouseY},
-			Width:    30,
-			Height:   64,
-		}
-
-		texture := engo.Files.Image("city.png")
-		city.RenderComponent = engo.NewRenderComponent(
-			texture,
-			engo.Point{0.5, 0.5},
-			"city texture",
-		)
+		// Create a new city
+		city := createCity(cb)
 
+		// Add to the system
 		for _, system := range cb.world.Systems() {
 			switch sys := system.(type) {
-			case *engo.RenderSystem:
+			case *common.RenderSystem:
 				sys.Add(&city.BasicEntity, &city.RenderComponent, &city.SpaceComponent)
 			}
 		}
 	}
 }
+
+// Remove is called whenever an Entity is removed from the scene, and thus from this system
+func (*CityBuildingSystem) Remove(ecs.BasicEntity) {}
+
+// ---------------------------------
+// Private functions
+
+// createCity creates a city
+func createCity(cb *CityBuildingSystem) (city City) {
+	// Set textures
+	texture, err := common.PreloadedSpriteSingle(CityAssets[0])
```

Comment:
Is there a real need for the CityAssets variable? What about just const `cityTextureLoction`. And if we did have a CityAssets variable it could be constant? It never changes.